code/mob/living.dm

```diff
@@ -385,8 +385,13 @@
 	if (!QDELETED(W) && (equipped() == W || usingInner))
 		var/pixelable = isturf(target)
 		if (!pixelable)
-			if (istype(target, /atom/movable) && isturf(target.loc))
```

Comment:
Okay I looked into this because this solution is still very oddly specific for the case we're handling here. The problem seems to be that the code assumes we won't ever be able to click on something which is inside something else unless it's in our inventory. So a better solution might be just replacing the `isturf(target.loc)` with `(isturf(target.loc) || !reach)`, that way if we can't reach something we will always trigger the pixel action and cancel the `afterattack`.